### DISCUSSION TITLE
Can anyone explain what header.isPlaceholder() is doing?

### DISCUSSION THREAD

**Discussion by elliotgonzalez-lk** [TanStack/table#4828] [General] [answered]
Across all the examples provided in the documentation, header.isPlaceholder() is called when building the table headers.  The docs refer to this as a check to see if a header is a "placeholder header".  However, there is no mention of how to create a placeholder header anywhere in the docs.  This line of code is showing as uncovered code in my test runner across all of my apps using tanstack table.  I'd like to just remove it, but if I knew how to create a placeholder header, I can just write a test and get that covered.  Can anyone help?

**Comment by talatkuyuk** [comment, marked as answer]
\+ 1

The secret is lay down [here](https://github.com/TanStack/table/blob/main/packages/table-core/src/core/headers.ts#LL446C1-L453C8)
```js
if (isLeafHeader && headerToGroup.column.parent) {
  // The parent header is new
  column = headerToGroup.column.parent
} else {
  // The parent header is repeated
  column = headerToGroup.column
  isPlaceholder = true
}
```
I suppose, it is related with grouping the headers, and **it is an internal mechanism of the react-table, you don't need to set it explicitly.** Someone can explain better, that is my comment from the reading of the code.

**Reply by elliotgonzalez-lk** [reply]
Thank you.  This was helpful.  You were correct, the placeholder headers refer to the grouped headers.  I passed in columns with grouped headers to my test cases and those lines are all reading covered now.

**Reply by FlorysiakSimon** [reply]
Hey, can you provide an exemple of how you tested this ? i'm having the same coverage issue and i don't really understand how you fixed it
